lib/Doctrine/ORM/Query/Exec/AbstractSqlExecutor.php

```diff
@@ -60,4 +73,22 @@ public function removeQueryCacheProfile()
      * @return Result|int
      */
     abstract public function execute(Connection $conn, array $params, array $types);
+
+    /** @return list<string> */
+    public function __sleep(): array
+    {
+        /* Two reasons for this:
+           - we do not need to serialize the deprecated property, we can
+             rebuild the reference to the new property in __wakeup()
+           - not having the legacy property in the serialized data means the
+             serialized representation becomes compatible with 3.0.x, meaning
+             there will not be a deprecation warning about a missing property
+             when unserializing data */
+        return array_values(array_diff(array_keys((array) $this), ["\0*\0_sqlStatements"]));
```

Comment:
This weirdness can be understood by reading https://www.phpinternalsbook.com/php5/classes_objects/serialization.html

Reply:
I did once some PHP shenanigans by creating private/protected stdClass properties just for fun. :joy: